src/leveldbwrapper.cpp

```diff
@@ -87,3 +106,18 @@ bool CLevelDBWrapper::WriteBatch(CLevelDBBatch& batch, bool fSync) throw(leveldb
     HandleError(status);
     return true;
 }
+
+// Prefixed with null character to avoid collisions with other keys
+const std::string CLevelDBWrapper::OBFUSCATE_KEY_KEY = "\000obfuscate_key";
+
+const unsigned int CLevelDBWrapper::OBFUSCATE_KEY_NUM_BYTES = 8;
+
+/**
+ * Returns a string (consisting of 8 random bytes) suitable for use as an 
+ * obfuscating XOR key. 
+ */
+std::string CLevelDBWrapper::CreateObfuscateKey() const {
+    unsigned char buff[OBFUSCATE_KEY_NUM_BYTES];
+    GetRandBytes(buff, OBFUSCATE_KEY_NUM_BYTES);
+    return std::string(&buff[0], &buff[OBFUSCATE_KEY_NUM_BYTES]);
```

Comment:
@KyrosKrane being a C++ newbie myself, I screwed this up the first time around; @pstratem [gave me some guidance](https://github.com/bitcoin/bitcoin/pull/6650#discussion_r38889104) on the proper incantation... though upon reading the docs again, I very much agree that what you're saying *looks* correct, so I gave it a shot. Unfortunately, the compiler spat back a
```
leveldbwrapper.cpp:122:57: error: call of overloaded ‘basic_string(unsigned char*, const unsigned int&)’ is ambiguous
     return std::string(&buff[0], OBFUSCATE_KEY_NUM_BYTES);
```
On further investigation, I realized this is because `buff` is an `unsigned char` array, not a `char` array -- the two types are "unrelated" types in C++, meaning (apparently) one can't easily be substituted for another. I believe this invocation of the constructor falls into the (7) range invocation with `unsigned char*` being the templated type.

I think you might be onto something here though -- could be that windows is treating the implicit conversion from unsigned char * to string oddly (anyone else think that's plausible?). I'm going to look for a simpler, less confusing way to do this. Maybe we just keep `obfuscate_key` as an unsigned char array instead of a string.

Thanks again for helping to dig into this problem.